rs/tests/execution/general_execution_tests/nns_shielding.rs

```diff
@@ -134,13 +130,86 @@ pub fn mint_cycles_not_supported_on_application_subnet(env: TestEnv) {
         let after_balance = get_balance(&canister_id, &agent).await;
         assert!(
             after_balance < before_balance,
-            "expected {} < expected {}",
+            "expected {} < {}",
             after_balance,
             before_balance
         );
     });
 }
 
+fn setup_ucan_and_try_mint128(node: IcNodeSnapshot) -> (AgentError, u128, u128, String) {
+    let agent = node.build_default_agent();
+    let canister_id = node.get_last_canister_id_in_allocation_ranges();
+    block_on(async move {
+        let canister_id = UniversalCanister::new_with_cycles(&agent, canister_id, *INITIAL_CYCLES)
+            .await
+            .unwrap()
+            .canister_id();
+        let before_balance = get_balance(&canister_id, &agent).await;
+        let res = agent
+            .update(&canister_id, "update")
+            .with_arg(
+                wasm()
+                    .mint_cycles128(Cycles::from(10_000_000_000u128))
+                    .reply_data_append()
+                    .reply()
+                    .build(),
+            )
+            .call_and_wait()
+            .await
+            .expect_err("should not succeed");
+        let after_balance = get_balance(&canister_id, &agent).await;
+        (res, before_balance, after_balance, canister_id.to_string())
+    })
+}
+
+pub fn mint_cycles128_supported_only_on_cycles_minting_canister(env: TestEnv) {
+    let nns_node = env.get_first_healthy_nns_node_snapshot();
+    let canister_id = nns_node.get_last_canister_id_in_allocation_ranges();
+    // Check that 'canister_id' is not 'CYCLES_MINTING_CANISTER_ID'.
+    assert_ne!(canister_id, CYCLES_MINTING_CANISTER_ID.into());
```

Comment:
```suggestion
```
this is only used as the effective canister id and thus it doesn't matter whether this is equal to CMC or not; if you prefer to keep this check, then I'd suggest to move it to `setup_ucan_and_try_mint128` and apply it to the actual canister id returned by `UniversalCanister::new_with_cycles`